core/arch/arm/mm/sp_mem.c

```diff
@@ -82,9 +84,6 @@ int sp_mem_add_pages(struct mobj *mobj, unsigned int *idx,
 	if (ADD_OVERFLOW(*idx, num_pages, &n) || n > tot_page_count)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	if (!core_pbuf_is(CORE_MEM_NON_SEC, pa, num_pages * SMALL_PAGE_SIZE))
```

Comment:
We will need some way of checking that the security state of the `mobj_sp` matches the one of the physical pages represented.

Reply:
Ack